src/Lykke.Job.CandlesHistoryWriter.Services/HistoryMigration/TradesMigrationManager.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.Job.CandlesHistoryWriter.Core.Domain.Candles;
+using Lykke.Job.CandlesHistoryWriter.Core.Domain.HistoryMigration.HistoryProviders.TradesSQLHistory;
+using Lykke.Job.CandleHistoryWriter.Repositories.HistoryMigration.HistoryProviders.TradesSQLHistory;
+using Lykke.Job.CandlesProducer.Contract;
+using Constants = Lykke.Job.CandlesHistoryWriter.Services.Candles.Constants;
+
+namespace Lykke.Job.CandlesHistoryWriter.Services.HistoryMigration
+{
+    [UsedImplicitly]
+    public class TradesMigrationManager
+    {
+        private readonly ICandlesHistoryRepository _candlesHistoryRepository;
+        private readonly ILog _log;
+
+        private readonly string _sqlConnString;
+        private readonly int _sqlQueryBatchSize;
+
+        public TradesMigrationHealthReport Health;
+
+        public TradesMigrationManager(
+            ICandlesHistoryRepository candlesHistoryRepository,
+            ILog log,
+            string sqlConnString,
+            int sqlQueryBatchSize
+            )
+        {
+            _candlesHistoryRepository = candlesHistoryRepository;
+            _log = log;
+
+            _sqlConnString = sqlConnString;
+            _sqlQueryBatchSize = sqlQueryBatchSize;
+
+            Health = null;
+        }
+
+        public void Migrate(ITradesMigrationRequest request)
+        {
+            Task.Run(() => DoMigrateAsync(request).Wait());
```

Comment:
.Wait() better to replace .GetAwaiter()..GetResult(). Wait wraps exception to the aggregate exception, GetAwaiter().GetResult() don't